beacon-chain/core/blocks/withdrawals_test.go

```diff
@@ -191,3 +196,385 @@ func TestProcessBLSToExecutionChange(t *testing.T) {
+				Withdrawals: []*enginev1.Withdrawal{
+					fullWithdrawal(1, 3),
+				},
+			},
+			Control: control{
+				LastWithdrawalValidatorIndex: 1,
+				NextWithdrawalIndex:          4,
+				Balances:                     map[uint64]uint64{1: 0},
+			},
+		},
+		{
+			Args: args{
+				Name:                         "success one partial withdrawal",
+				NextWithdrawalIndex:          21,
+				LastWithdrawalValidatorIndex: 37,
+				PartialWithdrawalIndices:     []types.ValidatorIndex{7},
+				Withdrawals: []*enginev1.Withdrawal{
+					partialWithdrawal(7, 21),
+				},
+			},
+			Control: control{
+				LastWithdrawalValidatorIndex: 7,
+				NextWithdrawalIndex:          22,
+				Balances:                     map[uint64]uint64{7: maxEffectiveBalance},
+			},
+		},
+		{
+			Args: args{
+				Name:                         "success many full withdrawals",
+				NextWithdrawalIndex:          22,
+				LastWithdrawalValidatorIndex: 4,
+				FullWithdrawalIndices:        []types.ValidatorIndex{7, 19, 28, 1},
+				Withdrawals: []*enginev1.Withdrawal{
+					fullWithdrawal(7, 22), fullWithdrawal(19, 23), fullWithdrawal(28, 24),
+					fullWithdrawal(1, 25),
+				},
+			},
+			Control: control{
+				LastWithdrawalValidatorIndex: 1,
+				NextWithdrawalIndex:          26,
+				Balances:                     map[uint64]uint64{7: 0, 19: 0, 28: 0, 1: 0},
+			},
+		},
+		{
+			Args: args{
+				Name:                         "success many partial withdrawals",
+				NextWithdrawalIndex:          22,
+				LastWithdrawalValidatorIndex: 4,
+				PartialWithdrawalIndices:     []types.ValidatorIndex{7, 19, 28, 1},
+				Withdrawals: []*enginev1.Withdrawal{
+					partialWithdrawal(7, 22), partialWithdrawal(19, 23), partialWithdrawal(28, 24),
+					partialWithdrawal(1, 25),
+				},
+			},
+			Control: control{
+				LastWithdrawalValidatorIndex: 1,
+				NextWithdrawalIndex:          26,
+				Balances: map[uint64]uint64{
+					7:  maxEffectiveBalance,
+					19: maxEffectiveBalance,
+					28: maxEffectiveBalance,
+					1:  maxEffectiveBalance,
+				},
+			},
+		},
+		{
+			Args: args{
+				Name:                         "success many withdrawals",
+				NextWithdrawalIndex:          22,
+				LastWithdrawalValidatorIndex: 12,
+				FullWithdrawalIndices:        []types.ValidatorIndex{7, 19, 28},
+				PartialWithdrawalIndices:     []types.ValidatorIndex{2, 1, 89, 15},
+				Withdrawals: []*enginev1.Withdrawal{
+					partialWithdrawal(15, 22), fullWithdrawal(19, 23), fullWithdrawal(28, 24),
+					partialWithdrawal(89, 25), partialWithdrawal(1, 26), partialWithdrawal(2, 27),
+					fullWithdrawal(7, 28),
+				},
+			},
+			Control: control{
+				LastWithdrawalValidatorIndex: 7,
+				NextWithdrawalIndex:          29,
+				Balances: map[uint64]uint64{
+					7: 0, 19: 0, 28: 0,
+					2: maxEffectiveBalance, 1: maxEffectiveBalance, 89: maxEffectiveBalance,
+					15: maxEffectiveBalance,
+				},
+			},
+		},
+		{
+			Args: args{
+				Name:                         "success more than max fully withdrawals",
```

Comment:
We don't have more than max withdrawals here

Reply:
we do, the fully withdrawable indices are more than 16